src/target/riscv_debug.c

```diff
@@ -304,6 +304,30 @@ void riscv_dmi_init(riscv_dmi_s *const dmi)
 	do {
 		/* Read out the DM's status register */
 		uint32_t dm_status = 0;
+		uint32_t dm_control = 0;
+		bool dm_active = false;
+		static uint8_t counter = 0;
```

Comment:
Please moves these down to where they're actually used - `dm_control` inside the loop just before the `riscv_dmi_read()` call, `dm_status` down to its `riscv_dmi_read()` call, etc.